zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/impl/log/LogElement.kt

```diff
@@ -0,0 +1,128 @@
+package org.hexworks.zircon.internal.component.impl.log
+
+import org.hexworks.zircon.api.builder.data.TileBuilder
+import org.hexworks.zircon.api.component.Component
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.data.Size
+import org.hexworks.zircon.api.graphics.SubTileGraphics
+import org.hexworks.zircon.api.modifier.Delay
+import org.hexworks.zircon.api.modifier.Modifier
+import org.hexworks.zircon.internal.component.renderer.LogElementRenderContext
+import org.hexworks.zircon.internal.component.renderer.LogElementRenderResult
+import org.hexworks.zircon.internal.component.renderer.VisibleRenderArea
+
+sealed class LogElement(private val xPosition: Int) {
+    var modifiers: Set<Modifier> = setOf()
+    var screenPositionArea: ScreenPositionArea? = null
```

Comment:
This can be a `Size` instead (with `yLength` of `1`). Can we have a sensible default value (or use `Maybe`) here? One of Kotlin's best feature is life without `null`s. 😄